html/mycakeapp/src/Controller/AuctionController.php

```diff
@@ -163,22 +170,181 @@ public function home()
 	{
 		// 自分が出品したBiditemをページネーションで取得
 		$biditems = $this->paginate('Biditems', [
-			'conditions'=>['Biditems.user_id'=>$this->Auth->user('id')], 
+			'conditions' => ['Biditems.user_id' => $this->Auth->user('id')],
 			'contain' => ['Users', 'Bidinfo'],
-			'order'=>['created'=>'desc'],
-			'limit' => 10])->toArray();
+			'order' => ['created' => 'desc'],
+			'limit' => 10
+		])->toArray();
 		$this->set(compact('biditems'));
+
+		$biditems = $this->Biditems->find('all')->toArray(); //出品者
+		$buyer_status = $this->BuyerStatus->find('all')->toArray(); //落札者の情報
+		// 出品者を取得
+		if ($this->request->is('post')) {
+			for ($i = 0; $i < count($biditems); $i++) {
+				if ($biditems[$i]["id"] === intval($this->request->getData(['item'])) && $biditems[$i]["user_id"] === $this->Auth->user('id')) {
+					$biditems_id = $biditems[$i]['id'];
+					$biditems_is_sent = $biditems[$i]['is_sent'];
+					break;
+				}
+			}
+			// 落札者の情報が入力されているか確認
+			$information = false;
+			for ($j = 0; $j < count($buyer_status); $j++) {
+				if ($buyer_status[$j]['biditem_id'] === $biditems_id) {
+					$information = true;
+					break;
+				}
+			}
+			if ($information === true && $biditems_is_sent === false) {
+				$entity = $this->Biditems->get($biditems_id);
+				$this->Biditems->patchEntity($entity, ['is_sent' => true]);
+				$this->Biditems->save($entity);
+				if ($this->Biditems->save($entity)) {
+					$bidinfo_array = $this->Bidinfo->find('all')->toArray();
+					for ($i = 0; $i <= count($bidinfo_array); $i++) {
+						if ($bidinfo_array[$i]['biditem_id'] === $entity["id"]) {
+							$bidinfo_id = $bidinfo_array[$i]['id']; //bidinfoのid
+							break;
+						}
+					}
+					$bidmsg = $this->Bidmessages->newEntity();
+					$bidmsgs = ['bidinfo_id' => $bidinfo_id, 'user_id' => $entity["user_id"], 'message' => '出品者から商品が発送されました。'];
+					$bidmsg = $this->Bidmessages->patchEntity($bidmsg, $bidmsgs);
+					// Bidmessageに発送完了を保存
+					$this->Bidmessages->save($bidmsg);
+					return $this->redirect(['action' => 'msg', $bidinfo_array[$i]['id']]);
+				}
+			}
+		}
+		$ratings = $this->Ratings->find('all')->toArray();
+		$this->set(compact('ratings', 'buyer_status'));
+	}
+	// 落札者情報
+	public function buyerinfo()
+	{
+		// 落札者とログインユーザーが一緒か、商品idが落札の商品idと一緒か==================================
+		// 自分が落札したBidinfoをページネーションで取得
+		$bidinfos = $this->paginate('Bidinfo', [
+			'conditions' => ['Bidinfo.user_id' => $this->Auth->user('id')],
+			'contain' => ['Users', 'Biditems']
+		])->toArray();
+		$bidinfo = $this->Bidinfo->find('all')->toArray();
+		for ($j = 0; $j < count($bidinfo); $j++) {
+			//商品idが落札されているかチェック
+			if ($bidinfo[$j]['biditem_id'] === intval($_GET["biditem_id"])) {
+				$bidinfo_biditem_id = $bidinfo[$j]['biditem_id'];
+				break;
+			} elseif ($bidinfo[$j]['biditem_id'] !== intval($_GET["biditem_id"])) {
+				$bidinfo_biditem_id = false;
+			}
+		}
+		if ($bidinfo_biditem_id === false) {
+			return $this->redirect(['action' => 'index']);
+		}
```

Comment:
(参考)
hitしなかった場合の判定についてですが
$bidinfo_biditem_id に毎回falseを代入するのはしっくりこないので
僕ならこう書くかなあってやつです
issetを使うとよさそうですね
```suggestion
		$countBidinfo = count($bidinfo);
		for ($j = 0; $j < $countBidinfo; $j++) {
			//商品idが落札されているかチェック
			if ($bidinfo[$j]['biditem_id'] === intval($_GET["biditem_id"])) {
				$bidinfo_biditem_id = $bidinfo[$j]['biditem_id'];
				break;
			}
		}
		if (false === isset($bidinfo_biditem_id)) {
			return $this->redirect(['action' => 'index']);
		}
```